homeassistant/components/hydrawise/config_flow.py

```diff
@@ -22,25 +24,61 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hydrawise."""
 
     VERSION = 1
-
-    async def _create_entry(
-        self, api_key: str, *, on_failure: Callable[[str], FlowResult]
+    MINOR_VERSION = 2
+
+    def __init__(self) -> None:
+        """Construct a ConfigFlow."""
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def _create_or_update_entry(
+        self,
+        api_key: str | None = None,
+        username: str | None = None,
+        password: str | None = None,
+        *,
+        on_failure: Callable[[str], FlowResult],
     ) -> FlowResult:
         """Create the config entry."""
-        api = legacy.LegacyHydrawiseAsync(api_key)
+
+        # Verify that the provided credentials work."""
+        if api_key:
+            api = legacy.LegacyHydrawiseAsync(api_key)
+        else:
+            api = client.Hydrawise(auth.Auth(username, password))
         try:
             # Skip fetching zones to save on metered API calls.
```

Comment:
The GraphQL API isn't metered so we don't need to pass `fetch_zones=False` anymore.